### PR TITLE
Fixes #25782 - better error when promoting conflicting docker repos

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: app/lib/katello/validators/environment_docker_repositories_validator.rb [hunk 1]
```diff
@@ -20,19 +20,21 @@ def validate(environment)
...
+          name_to_repos[name] ||= []
+          name_to_repos[name] << repository
         end
 
-        if names.length != names.uniq.length
-          error_messages << N_("Registry name pattern results in duplicate container image names")
+        duplicate_repos = name_to_repos.select { |_name, repos| repos.count > 1 }.values.flatten
+        if duplicate_repos.any?
+          repo_names = duplicate_repos.map(&:name).join(', ')
+          error_messages << N_("Registry name pattern results in duplicate container image names for these repositories: %s") % repo_names
```
Comment (mirzal) [inline app/lib/katello/validators/environment_docker_repositories_validator.rb]: Can we add "Consider changing the Lifecycle Environment's Registry Name Pattern to something more specific." at the end of this message?

Right now if I try to promote second CV, I get nice, detailed error message. If I try to promote one CV with two repos, I get error that tells me what is wrong (improvement over what we had!), but doesn't point me in direction to how I should fix it.

Comment (jlsherrill) [reply app/lib/katello/validators/environment_docker_repositories_validator.rb]: Error should now be:

```Registry name pattern results in duplicate container image names for these repositories: docker, docker2. Consider changing the Lifecycle Environment's Registry Name Pattern to something more specific.```

Comment (mirzal) [reply app/lib/katello/validators/environment_docker_repositories_validator.rb]: Confirmed. This is the message I see now. Thanks!

**Comment by sjha4** [comment]
@jlsherrill: What is the difference between N_() and _() in translation?

**Comment by jlsherrill** [comment]
@sjha4 N_() marks the string for translation, but does not return the translated string.  Its usually used in validations when you want the string to be translated, but you don't want the decision of what language to translate it in to come until its displayed to the user

**Pull request merged**